### DISCUSSION TITLE
Dropdown

### DISCUSSION THREAD

**Discussion by Qhk25** [primefaces/community#2447] [PrimeNG]
`<p-dropdown 
      #roleDropdown
      [options]="roles"
      optionLabel="text" 
      optionGroupLabel="value" 
      [filter]="true"
      filterBy="text"
      [dropdownIcon]="rolesLoadingIcon"  
      [style]="{width:'100%'}" 
      [appendTo]="'body'" 
      [showClear]="true" 
      formControlName="role"
      [lazy]="true"  
      [virtualScroll]="true"
      [virtualScrollItemSize]="50" 
      [virtualScrollOptions]="getOptions(userService.getRolesLists, roles)"             
      (onShow)="onDropdownShow($event, roles)"                                           
      placeholder="Select">
      </p-dropdown>`

` stopLoading = false;
  isLoading = false;
  getOptions(serviceName: any, targetArray: any[]) {
    return {
      delay: 0,
      showLoader: true,
      lazy: true,
      onLazyLoad: this.onLazyLoad.bind(this, serviceName, targetArray)
    };
  }
  
  onLazyLoad(serviceMethod: Function, targetArray: any[], event: any) {
    if (this.isLoading || this.stopLoading) return;
    this.isLoading = true;
  
    // Calculate the current page
    const currentPage = Math.floor(targetArray.length / 10) + 1;
  
    serviceMethod.call(this.userService, currentPage, 10).subscribe({
      next: (res: any) => {
        if (res.length > 0) {
          targetArray.push(...res);
          // Check if there might be more data to load
          if (res.length < 10) {
            this.stopLoading = true;
          }
        } else {
          this.stopLoading = true;
        }
        this.isLoading = false;
        // PrimeNG dropdown will automatically update when the targetArray changes
      },
      error: (err: any) => {
        this.isLoading = false;
        console.error(err);
      }
    });
  }
  
  onDropdownShow(event: any, targetArray: any[]) {
    this.stopLoading = false;
    this.isLoading = false;
    targetArray.length = 0; // Clear the array
    this.onLazyLoad(this.userService.getRolesLists, targetArray, event);
  }
  `

Above given is my code but it's not working as expected. Need help to make it work properly! 

https://github.com/user-attachments/assets/c0a4763b-2c19-492c-abcd-203178a7250e

https://github.com/user-attachments/assets/793a4a6a-7e0e-4277-9916-f77c82883fba

**Comment by ibrahimkelly** [comment]
I didn't notice any dysfunction; what could be the issue?
